logic.js

```diff
@@ -392,33 +392,34 @@ function ScenarioList(scenarios)
     var scenariolist = {};
     scenariolist.ul = document.createElement("ul");
     scenariolist.ul.className = "selectionlist";
-    scenariolist.radios = [];
+    scenariolist.spinner = null;
     scenariolist.decks = {};
 
     for (var i = 0; i < scenarios.length; i++)
     {
         var scenario = scenarios[i];
-        var listitem = document.createElement("li");
-        var dom_dict = create_input("radio", "scenario", scenario.name, scenario.name);
-
-        listitem.appendChild(dom_dict.root);
-        scenariolist.ul.appendChild(listitem);
-        scenariolist.radios.push(dom_dict.input);
-        scenariolist.decks[scenario.name] = scenario.decks;
+        scenariolist.decks[scenario.number] = scenario.decks;
     }
 
+    var listitem = document.createElement("li");
+    listitem.innerText = "Select scenario number";
+    scenariolist.ul.appendChild(listitem);
+
+    var scenario_spinner = create_input("number", "scenario_number", "1", "");
+    scenario_spinner.input.min = 0;
+    scenario_spinner.input.max = scenarios.length + 1;

```

Comment:
Problem is that we're skipping scenario 55 (as it's a random scenario). I agree, this is not the cleanest solution